### PR TITLE
fix lines with ':' that end in comments causing auto indent

### PR DESCRIPTION
**Pull request by nathanfranke**
Writing `var x := 5 #` then a newline would previously cause an automatic indentation. Now this is fixed.

### CODE REVIEW

File: scene/gui/code_edit.cpp [hunk 1]
```diff
@@ -994,7 +994,7 @@ void CodeEdit::_new_line(bool p_split_current_line, bool p_above) {
 			}
 
 			/* Make sure this is the last char, trailing whitespace or comments are okay. */
-			if (should_indent && (!is_whitespace(c) && is_in_comment(cl, cc) == -1)) {
+			if (should_indent && (!is_whitespace(c) && is_in_comment(cl, line_col) == -1)) {
```
Comment (nathanfranke) [inline scene/gui/code_edit.cpp:997]: Check if the current column is a comment rather than the last column.

**Comment by nathanfranke** [comment]
`line_col` is incremented by 1 because `is_in_comment` does not consider `#` to be part of the comment. Is this a bug?

https://github.com/godotengine/godot/blob/master/scene/gui/code_edit.cpp#L2461

**Comment by Paulb23** [comment]
Any chance you could add a test for this use case?

> `line_col` is incremented by 1 because `is_in_comment` does not consider `#` to be part of the comment. Is this a bug?

That's tricky, to me it's not a bug, but not sure what more natural for others.

**Review by Paulb23 — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**